course5/homework06/string.cc

```diff
@@ -0,0 +1,150 @@
+#include "string.h"
+#include<memory>
+#include<algorithm>
+#include<cstring>
+
+String::String()
+{
+
+}
+
+String::String(const String& src)   //copy constructor
+{
+    size = src.length();
+    charbuffer = std::make_unique<char[]>(size+1);
+    //for (int i=0; i < size+1; ++i)
+    //{
+        //charbuffer[i] = src.charbuffer[i];
+    //}
+    //std::copy(&(src.charbuffer[0]), &(src.charbuffer[size+1]), &(charbuffer[0])); //because charbuffer is unique pointer, otherwise if it's a char*, I can just use charbuffer
```

Comment:
It's nice to see you experimenting with different solution, and I'm happy that you settled for the best one :)
But we have version control (git), and you can safely remove the commented-out code.